### PR TITLE
Add the link to the examples and the development user guide.

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

**Comment by ksivaman** [comment]
@nouiz Could you sign your commit? 
Additionally, your change looks awkward in the readme file, it would be better if you used `*` instead of `-` to do the bullets.

**Comment by nouiz** [comment]
I amended the commit.
How do you generates the doc locally?

**Comment by ksivaman** [comment]
Since this is just a readme change you can just go to the PR branch and look at the file. It still isn't fixed though :)

**Comment by nouiz** [comment]
If I go in the "File Changed" tab, I see the updated code.
Your comment is very close just after I pushed. My shell session is still open and display a succesful push.
So I guess it is just that when you looked, github UI wasn't updated.
Sometimes it takes a few minutes.

If you look again, it should show as updted.

File: README.rst [hunk 1]
```diff
@@ -189,11 +189,13 @@ You can also specify which framework bindings to build. The default is pytorch o
...
-----------
+User Guide and Examples
+-----------------------
 
-For examples, tutorials and API reference please refer to the
-`User Guide <https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/index.html>`_.
+For examples, tutorials and API reference please refer to:
+* `User Guide <https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/index.html>`_ for the last release.
+* `Development User Guide <https://nvidia.github.io/TransformerEngine/>`_ for the development version.
+* `Examples <https://github.com/NVIDIA/TransformerEngine/tree/main/examples>`_.
```
Comment (ksivaman) [inline README.rst:198]:
```suggestion
For examples, tutorials and API reference please refer to:

* `User Guide <https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/index.html>`_ for the last release.
* `Development User Guide <https://nvidia.github.io/TransformerEngine/>`_ for the development version.
* `Examples <https://github.com/NVIDIA/TransformerEngine/tree/main/examples>`_.
```

Comment (ksivaman) [reply README.rst:198]: Maybe this should help. My comment was regarding the bullets not showing in the README, and it all just is in a single line.

**Review by ksivaman — APPROVED** [review]
LGTM

**Pull request merged**